lake/oneshot_cls.py

```diff
@@ -104,24 +105,46 @@ def main():
     model = CLS(image_shape, config, device=device, writer=writer).to(device)
 
   if not pretrained_model_path:
-    background_loader = torch.utils.data.DataLoader(
-        datasets.Omniglot('./data', background=True, download=True, transform=image_tfms),
-        batch_size=config['pretrain_batch_size'], shuffle=True)
+    val_split = 0.175
+
+    dataset = datasets.Omniglot('./data', background=True, download=True, transform=image_tfms)
+
+    val_size = round(val_split * len(dataset))
+    train_size = len(dataset) - val_size
+
+    train_set, val_set = torch.utils.data.random_split(dataset, [train_size, val_size])
```

Comment:
Because `datasets.Omniglot` is a 'PyTorch dataset', and `random_split` expects a dataset as input.. you would need to create _another_ class to encompass both of these. 

A new class (or fn) that defines the dataset first, then uses the `random_split`, then returns the sets.